### PR TITLE
new branching model

### LINKED ISSUE #2155: move new 1.0 policy to development docs
https://github.com/borgbackup/borg/milestone/38 see milestone description

### PR DESCRIPTION
**Pull request by enkore**
Fixes #2155 

(When we have multiple backport labels we can just make them all the same colour, no real need for colour coding there, and keeps colours free [all 17 million of them :D])

### CODE REVIEW

File: docs/development.rst [hunk 1]
```diff
@@ -56,6 +45,53 @@ Some guidance for contributors:
...
+   the label for each PR you've backported.
+4. Make a PR of the backporting branch against the maintenance branch
+   for backport review
+
+.. rubric:: Historic model
+
+Previously (until release 1.0.10) Borg used a back-merging model where
+most minor changes and fixes where committed to a maintenance branch
+(eg. 1.0-maint), and the maintenance branch(es) were regularly merged
+back into the main development branch. This became more and more
```
Comment (enkore) [inline docs/development.rst]: "merging upwards" according to gitworkflows(7)

File: docs/development.rst [hunk 2]
```diff
@@ -56,6 +45,53 @@ Some guidance for contributors:
...
+From time to time a maintainer will backport the changes for a
+maintenance branch, typically before a release or if enough changes
+were collected:
+
+1. (Notify others that you're doing this to avoid duplicate work)
+2. Branch a backporting branch off the maintenance branch
+3. Cherry pick and backport the changes from each labelled PR, remove
+   the label for each PR you've backported.
+4. Make a PR of the backporting branch against the maintenance branch
+   for backport review
```
Comment (enkore) [inline docs/development.rst]: done

File: docs/development.rst [hunk 3]
```diff
@@ -56,6 +45,54 @@ Some guidance for contributors:
 
 - wait for review by other developers
 
+Branching model
+---------------
+
+Borg development happens on the ``master`` branch and uses GitHub pull
+requests (if you don't have GitHub or don't want to use it you can
+send smaller patches via email to one of the maintainers, see AUTHORS).
```
Comment (ThomasWaldmann) [inline docs/development.rst]: ... via the borgbackup mailing list to the maintainers).

File: docs/development.rst [hunk 4]
```diff
@@ -56,6 +45,54 @@ Some guidance for contributors:
...
+From time to time a maintainer will backport the changes for a
+maintenance branch, typically before a release or if enough changes
+were collected:
+
+1. Notify others that you're doing this to avoid duplicate work.
+2. Branch a backporting branch off the maintenance branch.
+3. Cherry pick and backport the changes from each labelled PR, remove
+   the label for each PR you've backported.
+4. Make a PR of the backporting branch against the maintenance branch
+   for backport review.
```
Comment (textshell) [inline docs/development.rst]: I think we want some kind of comment in the original PR too. Maybe just reference the backported PR from the backporting branch PR, so github automatically places that comment?

Comment (enkore) [inline docs/development.rst]: Yeah, that makes it easier to see when-what-who happened

**Pull request merged**